### PR TITLE
Standardize templates across examples

### PR DESCRIPTION
**Pull request by rhcarvalho**
Changing Django, Rails and Node.js examples.
Tested all examples/templates with creating a project for each, deploying and curl'ing.

@bparees PTAL

### CODE REVIEW

**Comment by rhcarvalho** [comment]
@ewolinetz @sspeiche PTAL

**Comment by sspeiche** [comment]
@rhcarvalho I made a number of these changes already, probably a good idea to rebase. Please I believe @bparees wants to avoid ImageStream and instead use ImageStreamTags

File: openshift/templates/nodejs.json [hunk 1]
```diff
@@ -171,21 +185,17 @@
   "parameters": [
     {
       "name": "SOURCE_REPOSITORY_URL",
-      "description": "The URL with your Node.js application source code",
+      "description": "The URL of the repository with your application source code",
       "value": "https://github.com/openshift/nodejs-ex.git"
     },
     {
-      "name": "FRONTEND_SERVICE_NAME",
```
Comment (sspeiche) [inline openshift/templates/nodejs.json]: What is the service name stuff removed? I think just missed due to timing of pr #5

**Comment by rhcarvalho** [comment]
@sspeiche I rebased. Sorry, I've actually moved away from parameterizing the FRONTEND_SERVICE_NAME. That's something I introduced earlier, but looking at all templates it seemed that it was not bringing that much value for the level of indirection.
Do you find it useful?

I'm fine with either ImageStream or ImageStreamTags, let's her from @bparees then and I can update the PRs.

**Comment by sspeiche** [comment]
@rhcarvalho I'm fine with getting rid of service name stuff...it just adds to the UI and not sure getting much value out of it

File: openshift/templates/nodejs.json (hunk 1 above)
Comment (bparees) [inline openshift/templates/nodejs.json]: yeah I think there's a reversion in this pr.

**Comment by bparees** [comment]
ok, hardcoding the service name is fine.

but we should be using imagestreamtags not imagestreams.

**Comment by rhcarvalho** [comment]
I can change that tomorrow. No computer here now.
The ImageStream references are all converted to ImageStreamTags by appending the ":latest" suffix, what's the reasoning to prefer one over the other in templates?

**Comment by bparees** [comment]
@rhcarvalho ImageStream is basically deprecated.  ImageStreamTag is cleaner, since ImageStream required a separate field to indicate the Tag.

**Comment by bparees** [comment]
@rhcarvalho also the imagestreamtags we define now have versions..you should use the version, not latest:

https://github.com/openshift/origin/blob/master/examples/image-streams/image-streams-centos7.json#L20

**Comment by rhcarvalho** [comment]
I'll move all examples to use ImageStreamTags tomorrow morning, and also add a DB template for node.js (green flag from you @sspeiche?).

@bparees, regarding versions, that makes sense for the image streams we provide in origin, the ones you linked to. But I didn't understand how that applies to the framework examples. So far we're defining ImageStreams in those templates for a out-of-box experience -- independent of some admin adding proper image streams to projects a priori.
Are you suggesting we change the IS definitions to match those in origin?

**Comment by bparees** [comment]
@rhcarvalho We're actually going to be removing the imagestreams that the templates define and rely entirely on an admin creating the imagestreams in the openshift namespace.  So the templates will no longer be "self contained".  i'll forward you the email thread.

**Comment by rhcarvalho** [comment]
Ok, ok... how about the image streams for application code, the ones we push builds to?

**Comment by bparees** [comment]
@rhcarvalho yes we'll still define an output stream.

**Comment by rhcarvalho** [comment]
@bparees I've updated rails, django and nodejs examples to use the image streams in the "openshift" namespace and use the version tags (`ruby:2.0`, `python:3.3`, `nodejs:0.10`)

**Comment by mfojtik** [comment]
@bparees @rhcarvalho would be nice to create an test in jenkins that start openshift and submit all these templates and check if the frontends are running...

**Comment by bparees** [comment]
@mfojtik yes it would :)

why not add it to extended tests?

File: openshift/templates/nodejs.json [hunk 2]
```diff
@@ -162,21 +173,17 @@
...
     {
       "name": "FRONTEND_ROUTE",
       "description": "The exposed hostname that will route to the Node.js service",
-      "value": "nodejs.apps"
-    }    
-  ],
-  "labels": {
-    "template": "nodejs-example"
-  }
+      "value": "nodejs.app"
```
Comment (bparees) [inline openshift/templates/nodejs.json]: make this look more like a hostname.

File: openshift/templates/nodejs.json [hunk 3]
```diff
@@ -4,17 +4,23 @@
   "metadata": {
     "name": "nodejs-example",
     "annotations": {
-      "description": "A simple Node.js application with no database",
-      "iconClass": "icon-nodejs",
-      "tags": "instant-app,nodejs"
+      "description": "An example Node.js application with no database",
+      "tags": "instant-app,nodejs",
+      "iconClass": "icon-nodejs"
```
Comment (mfojtik) [inline openshift/templates/nodejs.json]: make sure we have this icon ;-)

File: openshift/templates/nodejs.json [hunk 4]
```diff
@@ -55,67 +64,65 @@
...
-        }
+        },
+        "triggers": [
+          {
+            "type": "ImageChange"
+          },
+          {
+            "type": "GitHub",
+            "github": {
+              "secret": "secret123"
```
Comment (mfojtik) [inline openshift/templates/nodejs.json]: make this parameter

Comment (mfojtik) [inline openshift/templates/nodejs.json]: please make it so in every template. this is very important as the parameters are only way users can change this value when creating applications from UI

File: openshift/templates/nodejs.json [hunk 5]
```diff
@@ -55,67 +64,65 @@
...
+            }
+          },
+          {
+            "type": "Generic",
```
Comment (mfojtik) [inline openshift/templates/nodejs.json]: i'm not sure if generic hooks have sense here since the repository is github

Comment (bparees) [inline openshift/templates/nodejs.json]: agreed, let's remove it.

File: openshift/templates/nodejs.json (hunk 3 above)
Comment (rhcarvalho) [inline openshift/templates/nodejs.json]: @mfojtik we do: https://github.com/openshift/origin/blob/master/assets/app/styles/_openshift-logos-icon.less#L64

File: openshift/templates/nodejs.json (hunk 4 above)
Comment (rhcarvalho) [inline openshift/templates/nodejs.json]: done

File: openshift/templates/nodejs.json (hunk 5 above)
Comment (rhcarvalho) [inline openshift/templates/nodejs.json]: removed from all templates

File: openshift/templates/nodejs.json (hunk 2 above)
Comment (rhcarvalho) [inline openshift/templates/nodejs.json]: Please let's do this in a another group of PRs.

**Comment by bparees** [comment]
lgtm.

**Pull request merged**